packages/shared-preferences/android/src/main/java/io/flutter/plugins/shared_preferences/SharedPreferencesPlugin.java

```diff
@@ -59,7 +59,11 @@ public void onMethodCall(MethodCall call, MethodChannel.Result result) {
         result.success(null);
         break;
       case "setInt":
-        editor.putInt(key, (int) call.argument("value")).apply();
+        // According to platform-channels readme, ints can come as java.lang.Integer
+        // java.lang.Long or (if larger than 64 bits) java.math.BigInteger. We handle
+        // the first two easily by casting to long. Clients won't store larger than
+        // 64 bits here, will they?...
```

Comment:
Would it be worthwhile to assert on the dart side that values stay within bounds?